From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shane Freeder <theboyetronic@gmail.com>
Date: Tue, 3 Mar 2020 05:26:40 +0000
Subject: [PATCH] Prevent teleporting dead entities


diff --git a/src/main/java/net/minecraft/server/network/PlayerConnection.java b/src/main/java/net/minecraft/server/network/PlayerConnection.java
index d150a2d0a0c0b28d701d64b7a7eaa6adbff9dc3c..0fa39f4f9a90a2fcbd5dd8d9d33f8d912b5c9e91 100644
--- a/src/main/java/net/minecraft/server/network/PlayerConnection.java
+++ b/src/main/java/net/minecraft/server/network/PlayerConnection.java
@@ -1467,6 +1467,10 @@ public class PlayerConnection implements PacketListenerPlayIn {
     }
 
     private void internalTeleport(double d0, double d1, double d2, float f, float f1, Set<PacketPlayOutPosition.EnumPlayerTeleportFlags> set) {
+        if (player.dead) {
+            LOGGER.info("Attempt to teleport dead player {} restricted", player.getName());
+            return;
+        }
         // CraftBukkit start
         if (Float.isNaN(f)) {
             f = 0;
